libs/feature/record/src/lib/links/link-utils.ts

```diff
@@ -47,7 +47,12 @@ export function getLinksWithWfsFormats(
   link: MetadataLinkValid
 ): Promise<MetadataLinkValid[]> {
   return new WfsEndpoint(link.url).isReady().then((endpoint) => {
-    const featureType = endpoint.getFeatureTypeSummary(link.name)
+    let featureType = endpoint.getFeatureTypeSummary(link.name)
+    if (!featureType) {
+      //fallback: if link name does not match layer name, use first layer
+      featureType = endpoint.getFeatureTypes()[0]
+      featureType.outputFormats = endpoint.getServiceInfo().outputFormats
```

Comment:
Why is this needed? this should be the responsibility of ogc-client I think

Reply:
You are right that this should perhaps rather be handled by ogc-client. One example is https://dservices8.arcgis.com/rxZzohbySMKHTNcy/arcgis/services/ind_hdf/WFSServer?SERVICE=WFS&REQUEST=GetCapabilities (which you find by searching for "ind hdf" on datahub). Here the `link.name` in the metadata is `ArcGIS GeoService` whereas the `FeatureType` name is `ind_hdf:ind_hdf_2021`. Hum, just noticing, that this PR does not work on the gh-pages deployment :-(

Reply:
Ok so the feature type seems to be the hard to guess based on the record metadata so this probably cannot be handled consistently in the JSON harvester. On the other hand I'm not a big fan of a catch-all fallback logic like this because it produces inconsistent behaviour. For example if a record points to a dataset in a WFS endpoint but for whatever reason this dataset disappeared and there's only one other completely unrelated dataset left, then we'll show this unrelated dataset and that would not make any sense.

I think what we should do is if the feature type name does not match then too bad, throw an error and move on. We can improve the JSON harvester to "guess" the feature type name which apparently would work in many cases, or we could also assume that if a `WFSServer` is available then a `FeatureServer` will be as well (to be discussed with the platform maintainers maybe?).

What do you think?

Reply:
> Hum, just noticing, that this PR does not work on the gh-pages deployment :-(

No worries, that's because the deployed app points to gn4.georchestra.org which does not have up to date harvesters for ESRI.

Reply:
Yes, I agree, that this fallback might lead to unexpected behavior as you described. Let's go for a clear error handling, then!

Looking at `https://data-atmo-hdf.opendata.arcgis.com/data.json` we cannot assume that data with a `WFSServer` is often also available via a `FeatureServer`, but I don't think this would help anyways, as `FeatureServer` metadata does not give better information and we do not consume these services for data.